spec/requests/authentications_spec.rb

```diff
@@ -0,0 +1,107 @@
+require 'rails_helper'
+require 'openid_connect'
+require 'json'
+
+RSpec.describe "Authentications", type: :request do
+  before() do
+    allow(ForestLiana::IpWhitelist).to receive(:retrieve) { true }
+    allow(ForestLiana::IpWhitelist).to receive(:is_ip_whitelist_retrieved) { true }
+    allow(ForestLiana::IpWhitelist).to receive(:is_ip_valid) { true }
+    allow(ForestLiana::OidcConfigurationRetriever).to receive(:retrieve) {
+      JSON.parse('{
+          "registration_endpoint": "https://api.forestadmin.com/oidc/registration",
+          "issuer": "api.forestadmin.com"
+        }', :symbolize_names => false)
+    }
+    allow(ForestLiana::ForestApiRequester).to receive(:post) {
+      instance_double(HTTParty::Response, body: '{ "client_id": "random_id" }', code: 201)
+    }
+    allow_any_instance_of(OpenIDConnect::Client).to receive(:access_token!) {
+      OpenIDConnect::AccessToken.new(access_token: 'THE-ACCESS-TOKEN', client: instance_double(OpenIDConnect::Client))
+    }
+  end
+
+  after() do
```

Comment:
[MISC] Unneeded parenthesis
```suggestion
  after do
```